src/java/org/apache/cassandra/index/sai/disk/v1/VectorIndexSearcher.java

```diff
@@ -103,7 +109,8 @@ private PostingList searchPosting(QueryContext context, Expression exp, Abstract
             return bitsOrPostingList.postingList();
 
         float[] queryVector = exp.lower.value.vector;
-        return graph.search(queryVector, limit, bitsOrPostingList.getBits(), Integer.MAX_VALUE, context);
+        return graph.search(queryVector, limit, bitsOrPostingList.getBits(), Integer.MAX_VALUE, context,
+                            new SSTableRowIdToScoreCache(sstableId, context, metadata.segmentRowIdOffset));
```

Comment:
(1) This syntax obscures the fact that the cache is actually "owned" by the QueryContext (and SSTRITSC is just a wrapper that includes the sstable id).  Can we make move this to a static method of QueryContext?

(2) The name of the class should also reflect that the actual cache is the Map in the QC (and thus the lifetime is longer than just a single graph search which would not be super useful)

Reply:
ScoreCacheManager or ScoreCacheProxy, maybe

Reply:
I completely agree about the name. I'll revisit it based on these comments.